scripts/pre_commit_linter.py

```diff
@@ -1478,6 +1478,103 @@ def _check_directive_scope(all_files, parsed_js_files):
+                if parsed_node['type'] != 'ExpressionStatement':
+                    continue
+                # Separate the expression part of the node.
+                expression = parsed_node['expression']
+                # Check whether the expression belongs to a directive.
+                expression_type_is_not_call = (
+                    expression['type'] != 'CallExpression')
+                if expression_type_is_not_call:
+                    continue
+                expression_callee_type_is_not_member = (
+                    expression['callee']['type'] != 'MemberExpression')
+                if expression_callee_type_is_not_member:
+                    continue
+                callee_property = expression['callee']['property']['name']
+                if callee_property != 'factory' and (
+                        callee_property != 'directive' and
+                        callee_property != 'controller' and
+                        callee_property != 'filter'):
+                    continue
+                propertyz = callee_property
+                # Separate the arguments of the expression.
+                arguments = expression['arguments']
+                # The first argument of the expression is the
+                # name of the property.
+                property_name = arguments[0]['value']
+                property_num += 1
+                if callee_property == 'directive':
+                    if (property_name[0].upper() + property_name[1:] +
+                            'Directive' != (exact_filename)):
+                        print (
+                            '%s -> Please ensure that the %s name strictly '
+                            'matches the filename'
+                            % (filename, callee_property))
+                        failed = True
+                elif callee_property == 'filter':
+                    if (property_name[0].upper() + property_name[1:] +
+                            'Filter' != (exact_filename)):
+                        print (
+                            '%s -> Please ensure that the %s name strictly '
+                            'matches the filename'
+                            % (filename, callee_property))
+                        failed = True
+                else:
+                    if property_name != exact_filename:
+                        print (
+                            '%s -> Please ensure that the %s name strictly '
+                            'matches the filename'
+                            % (filename, callee_property))
+                        failed = True
+
+            if property_num > 1:
+                print (
+                    '%s -> Please ensure that there is exactly one %s in the'
+                    ' file.' % (filename, propertyz))
+                failed = True
+
+    with _redirect_stdout(_TARGET_STDOUT):
+        if failed:
+            summary_message = '%s  Js property check failed' % (
```

Comment:
I have no clue what this means, you should use a message that is clearer for newcomer contributors.

Reply:
Done